src/Uno.UI/Microsoft/UI/Xaml/Controls/TabView/TabView.cs

```diff
@@ -0,0 +1,1248 @@
+			var listView = m_listView;
+			if (listView != null)
+			{
+				SelectedIndex = listView.SelectedIndex;
+				SelectedItem = listView.SelectedItem;
+			}
+
+			UpdateTabContent();
+
+			SelectionChanged?.Invoke(this, args);
+		}
+
+		TabViewItem FindTabViewItemFromDragItem(object item)
+		{
+			var tab = ContainerFromItem(item) as TabViewItem;
+
+			if (tab == null)
+			{
+				var fe = item as FrameworkElement;
+				if (fe != null)
+				{
+					tab = VisualTreeHelper.GetParent(fe) as TabViewItem;
+				}
+			}
+
+			if (tab == null)
+			{
+				// This is a fallback scenario for tabs without a data context
+				var numItems = TabItems.Count;
+				for (int i = 0; i < numItems; i++)
+				{
+					var tabItem = ContainerFromIndex(i) as TabViewItem;
+					if (tabItem.Content == item)
+					{
+						tab = tabItem;
+						break;
+					}
+				}
+			}
+
+			return tab;
+		}
+
+		private void OnListViewDragItemsStarting(object sender, DragItemsStartingEventArgs args)
+		{
+			var item = args.Items[0];
+			var tab = FindTabViewItemFromDragItem(item);
+			var myArgs = new TabViewTabDragStartingEventArgs(args.Data, item, tab);
+
+			TabDragStarting?.Invoke(this, myArgs);
+		}
+
+		private void OnListViewDragOver(object sender, DragEventArgs args)
+		{
+			TabStripDragOver?.Invoke(this, args);
+		}
+
+		void OnListViewDrop(object sender, DragEventArgs args)
+		{
+			TabStripDrop?.Invoke(this, args);
+		}
+
+		private void OnListViewDragItemsCompleted(object sender, DragItemsCompletedEventArgs args)
+		{
+			var item = args.Items[0]; //TODO:MZ: Needs GetAt?
+			var tab = FindTabViewItemFromDragItem(item);
+			var myArgs = new TabViewTabDragCompletedEventArgs(args.DropResult, item, tab);
+
+			TabDragCompleted?.Invoke(this, myArgs);
+
+			// None means it's outside of the tab strip area
+			if (args.DropResult == DataPackageOperation.None)
+			{
+				var tabDroppedArgs = new TabViewTabDroppedOutsideEventArgs(item, tab);
+				TabDroppedOutside?.Invoke(this, tabDroppedArgs);
+			}
+		}
+
+		void UpdateTabContent()
+		{
+			var tabContentPresenter = m_tabContentPresenter;
+			if (tabContentPresenter != null)
+			{
+				if (SelectedItem == null)
+				{
+					tabContentPresenter.Content = null;
+					tabContentPresenter.ContentTemplate = null;
+					tabContentPresenter.ContentTemplateSelector = null;
+				}
+				else
+				{
+					var tvi = SelectedItem as TabViewItem;
+					if (tvi == null)
+					{
+						tvi = ContainerFromItem(SelectedItem) as TabViewItem;
+					}
+
+					if (tvi != null)
+					{
+						// If the focus was in the old tab content, we will lose focus when it is removed from the visual tree.
+						// We should move the focus to the new tab content.
+						// The new tab content is not available at the time of the LosingFocus event, so we need to
+						// move focus later.
+						bool shouldMoveFocusToNewTab = false;
+						var revoker = tabContentPresenter.LosingFocus(auto_revoke, [&shouldMoveFocusToNewTab](object sender, LosingFocusEventArgs args)
```

Comment:
Codacy found an issue: [Remove the unused local variable 'revoker'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6063676)